themes/theme-ios11/pages/Product/components/AddToCartBar/index.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { Fragment, Component } from 'react';
+import PropTypes from 'prop-types';
+import UIEvents from '@shopgate/pwa-core/emitters/ui';
+import Portal from '@shopgate/pwa-common/components/Portal';
+import {
+  PRODUCT_ADD_TO_CART_BAR,
+  PRODUCT_ADD_TO_CART_BAR_AFTER,
+  PRODUCT_ADD_TO_CART_BAR_BEFORE,
+} from '@shopgate/pwa-common-commerce/product/constants/Portals';
+import * as constants from './constants';
+import AddToCartButton from './components/AddToCartButton';
+import AddMoreButton from './components/AddMoreButton';
+import CartItemsCount from './components/CartItemsCount';
+import connect from './connector';
+import styles from './style';
+
+/**
+ * The AddToCartBar component.
+ */
+class AddToCartBar extends Component {
+  static propTypes = {
+    conditioner: PropTypes.shape().isRequired,
+    options: PropTypes.shape().isRequired,
+    productId: PropTypes.string.isRequired,
+    addToCart: PropTypes.func,
+    disabled: PropTypes.bool,
+    loading: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    addToCart: () => {},
+    disabled: false,
+    loading: false,
+  };
+
+  /**
+   * Constructor.
+   */
+  constructor() {
+    super();
+
+    this.state = {
+      clicked: false,
+      visible: true,
+      added: 0,
+    };
+
+    UIEvents.addListener(constants.SHOW_ADD_TO_CART_BAR, this.handleShow);
+    UIEvents.addListener(constants.HIDE_ADD_TO_CART_BAR, this.handleHide);
+    UIEvents.addListener(constants.INCREMENT_ACTION_COUNT, this.handleIncrement);
+    UIEvents.addListener(constants.DECREMENT_ACTION_COUNT, this.handleDecrement);
+    UIEvents.addListener(constants.RESET_ACTION_COUNT, this.handleReset);
+  }
+
+  /**
+   * Unregisters the ui events.
+   */
+  componentWillUnmount() {
+    UIEvents.removeListener(constants.SHOW_ADD_TO_CART_BAR, this.handleShow);
+    UIEvents.removeListener(constants.HIDE_ADD_TO_CART_BAR, this.handleHide);
+    UIEvents.removeListener(constants.INCREMENT_ACTION_COUNT, this.handleIncrement);
+    UIEvents.removeListener(constants.DECREMENT_ACTION_COUNT, this.handleDecrement);
+    UIEvents.removeListener(constants.RESET_ACTION_COUNT, this.handleReset);
+  }
+
+  handleShow = () => {
+    this.setState({ visible: true });
+  }
+
+  handleHide = () => {
+    this.setState({ visible: false });
+  }
+
+  handleIncrement = () => {
+    this.setState(prevState => ({
+      added: (prevState.added + 1),
+    }));
+  }
+
+  handleDecrement = () => {
+    this.setState(prevState => ({
+      added: (prevState.added > 0 ? prevState.added - 1 : 0),
+    }));
+  }
+
+  handleReset = () => {
+    this.setState({ added: 0 });
+  }
+
+  /**
+   * Handles the button click.
+   * Checks if the button can be clicked and if
+   * all criteria set by the conditioner are met.
+   */
+  handleAddToCart = () => {
+    if (this.state.clicked) {
```

Comment:
When you add a single product to the cart, switch to the cart and go back, you can't put further products to the cart. That's because the clicked state is not reset.

Reply:
fixed